### PR TITLE
Swaps: Allow every token on user wallet to be swapped

### LINKED ISSUE #2699: Swaps: Allow every token on my wallet to be swapped
Currently the Swaps flow and token search results is limited to the tokens that are present on the Swaps API metadata.

As a user, I want to be able to attempt swapping any token present on my wallet

### PR DESCRIPTION
**Pull request by wachunei**
**Description**

Currently the Swaps flow and token search results is limited to the tokens that are present on the Swaps API metadata.

As a user, I want to be able to attempt swapping any token present on my wallet

**Issue**

Resolves #2699

### CODE REVIEW

File: app/reducers/swaps/index.js [hunk 1]
```diff
@@ -56,9 +56,34 @@ export const swapsHasOnboardedSelector = createSelector(
...
 const swapsControllerTokens = state => state.engine.backgroundState.SwapsController.tokens;
+const tokensSelectors = state => state.engine.backgroundState.AssetsController.tokens;
+
+const swapsControllerAndUserTokens = createSelector(
+	swapsControllerTokens,
+	tokensSelectors,
+	(swapsTokens, tokens) => {
+		const values = [...(swapsTokens || []), ...(tokens || [])]
+			.reduce((map, token) => {
+				const key = token.address.toLowerCase();
```
Comment (rickycodes) [inline app/reducers/swaps/index.js:67]: Is there ever an instance where `token.address` might be undefined?

**Review by rickycodes — APPROVED** [review]
one small question but otherwise LGTM

File: app/components/UI/Swaps/QuotesView.js [hunk 2]
```diff
@@ -216,7 +216,7 @@ async function resetAndStartPolling({ slippage, sourceToken, destinationToken, s
 	// ff the token is not in the wallet, we'll add it
 	if (
 		!isSwapsNativeAsset(destinationToken) &&
-		!contractExchangeRates[safeToChecksumAddress(destinationToken.address)]
+		!(safeToChecksumAddress(destinationToken.address) in contractExchangeRates)
 	) {
 		const { address, symbol, decimals } = destinationToken;
 		await AssetsController.addToken(address, symbol, decimals);
```
Comment (wachunei) [inline app/components/UI/Swaps/QuotesView.js:222]: This was adding the token every time even if the exchange rate was 0

File: app/reducers/swaps/index.js (hunk 1 above)
Comment (wachunei) [reply app/reducers/swaps/index.js:67]: should never happen 👀

**Review by ibrahimtaveras00 — APPROVED** [review]
QA Passed 👍🏽

**Pull request merged**